x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/XPackPluginTests.java

```diff
@@ -85,19 +132,119 @@ public void testNodesNotReadyForXPackCustomMetadata() {
         }
     }
 
-    private XPackPlugin createXPackPlugin(Settings settings) throws Exception {
-        return new XPackPlugin(settings) {
+    public void testLoadExtensions() throws Exception {
+        XPackPlugin xpackPlugin = createXPackPlugin(Settings.builder().build());
+        xpackPlugin.loadExtensions(new ExtensiblePlugin.ExtensionLoader() {
+            @Override
+            @SuppressWarnings("unchecked")
+            public <T> List<T> loadExtensions(Class<T> extensionPointType) {
+                List<Object> extensions = new ArrayList<>();
+                if (extensionPointType == MutableLicenseService.class) {
+                    extensions.add(new TestLicenseService());
+                }
+                return (List<T>) extensions;
+            }
+        });
+        assertEquals(license, XPackPlugin.getSharedLicenseService().getLicense());
+        assertEquals(License.OperationMode.resolve(licenseType), XPackPlugin.getSharedLicenseState().getOperationMode());
```

Comment:
I think we should also call `createComponents` and ensure the licenseService stays as the same.

Reply:
good call... added a call to the method in [`05bdab9` (#94927)](https://github.com/elastic/elasticsearch/pull/94927/commits/05bdab9d211bebf7e7d2c1fb92d16c111fc55dab)